File: app/src/main/java/com/wmods/wppenhacer/xposed/features/general/DeleteStatus.java
package com.wmods.wppenhacer.xposed.features.general;


import android.view.Menu;
import android.view.MenuItem;

import androidx.annotation.NonNull;

import com.wmods.wppenhacer.R;
import com.wmods.wppenhacer.xposed.core.Feature;
import com.wmods.wppenhacer.xposed.core.WppCore;
import com.wmods.wppenhacer.xposed.core.components.FMessageWpp;
import com.wmods.wppenhacer.xposed.core.db.MessageStore;
import com.wmods.wppenhacer.xposed.core.devkit.Unobfuscator;
import com.wmods.wppenhacer.xposed.features.listeners.MenuStatusListener;

import org.luckypray.dexkit.query.enums.StringMatchType;

import java.util.List;

import de.robv.android.xposed.XSharedPreferences;

public class DeleteStatus extends Feature {


    public DeleteStatus(@NonNull ClassLoader classLoader, @NonNull XSharedPreferences preferences) {
        super(classLoader, preferences);
    }

    @Override
    public void doHook() throws Throwable {

        var StatusPlaybackActivityClass = Unobfuscator.findFirstClassUsingName(classLoader, StringMatchType.EndsWith, "StatusPlaybackActivity");

        var item = new MenuStatusListener.OnMenuItemStatusListener() {

            @Override
            public MenuItem addMenu(Menu menu, List<FMessageWpp> fMessageWppList, int currentIndex) {
                if (menu.findItem(R.string.delete_for_me) != null) return null;
                var fMessage = fMessageWppList.get(currentIndex);
                if (fMessage.getKey().isFromMe) return null;
                return menu.add(0, R.string.delete_for_me, 0, R.string.delete_for_me);
            }

            @Override
            public void onClick(MenuItem item, Object fragmentInstance, List<FMessageWpp> fMessageWppList, int currentIndex) {
                try {
                    var key = fMessageWppList.get(currentIndex).getKey();
                    MessageStore.getInstance().deleteStatusByMessageKey(key.messageID);
                    var activity = WppCore.getCurrentActivity();
                    if (activity != null && StatusPlaybackActivityClass.isInstance(activity)) {
                        activity.recreate();
                    }
                } catch (Exception e) {
                    logDebug(e);
                }
            }
        };
        MenuStatusListener.getMenuStatuses().add(item);
    }

    @NonNull
    @Override
    public String getPluginName() {
        return "Delete Status";
    }
}

File: app/src/main/java/com/wmods/wppenhacer/xposed/features/general/DeleteStatus.kt
package com.wmods.wppenhacer.xposed.features.general

import android.view.Menu
import android.view.MenuItem
import com.wmods.wppenhacer.R
import com.wmods.wppenhacer.xposed.core.Feature
import com.wmods.wppenhacer.xposed.core.WppCore
import com.wmods.wppenhacer.xposed.core.db.MessageStore
import com.wmods.wppenhacer.xposed.core.devkit.Unobfuscator
import com.wmods.wppenhacer.xposed.features.listeners.MenuStatusListener
import com.wmods.wppenhacer.xposed.utils.Utils
import android.content.SharedPreferences
import android.widget.Toast
import org.luckypray.dexkit.query.enums.StringMatchType

class DeleteStatus(classLoader: ClassLoader, preferences:SharedPreferences) : Feature(classLoader, preferences) {

    @Throws(Throwable::class)
    override fun doHook() {
        val statusPlaybackActivityClass = Unobfuscator.findFirstClassUsingName(classLoader, StringMatchType.EndsWith, "StatusPlaybackActivity")

        val item = object : MenuStatusListener.OnMenuItemStatusListener() {

            override fun addMenu(menu: Menu, statusData: MenuStatusListener.StatusData): MenuItem? {
                if (menu.findItem(R.string.delete_for_me) != null) return null
                if (statusData.currentItem.isFromMe) return null
                return menu.add(0, R.string.delete_for_me, 0, R.string.delete_for_me)
            }

            override fun onClick(item: MenuItem, statusData: MenuStatusListener.StatusData) {
                val activity = WppCore.getCurrentActivity()
                val messageId = statusData.currentItem.messageID

                MessageStore.getInstance().deleteStatusByMessageKey(messageId) { success ->
                    if (success && activity != null && statusPlaybackActivityClass.isInstance(activity)) {
                        activity.runOnUiThread {
                            if (activity.isFinishing || activity.isDestroyed) return@runOnUiThread
                            val itemList = statusData.getCurrentItemList()
                            val isLastItem = statusData.currentIndex >= itemList.size - 1

                            if (itemList.size <= 1 || isLastItem) {
                                activity.finish()
                            } else {
                                activity.recreate()
                                activity.overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
                            }
                        }
                    }
                }
            }
        }
        MenuStatusListener.menuStatuses.add(item)
    }

    override fun getPluginName(): String {
        return "Delete Status"
    }
}
